### PR TITLE
V4 with Python 3.6, PuppetDBs < 5.2 support drop and Flask (and more) upgrade

### PR DESCRIPTION
**Pull request by gdubicki**
With a recent [first release of Requests that drops support for Python 3.6](https://github.com/psf/requests/releases/tag/v2.28.0) I realized that we should also drop it in the Puppetboard or else we will not be able to apply possible future security updates. Python 3.6 had its EOL in December 2021 so we already supported it for about 1/2 year extra.

The only well-known to me and officially maintained system that doesn't have Python 3.7+ is Centos 7, but for that I am hoping to provide a solution using Python 3.8 and HTTPD 2.4 from the open-suorce version of [Red Hat Software Collections 3.8 ](https://access.redhat.com/documentation/en-us/red_hat_software_collections/3/html/3.8_release_notes/chap-rhscl) downloaded from https://www.softwarecollections.org/en/. Or alternatively you can just use Docker.

As for the old PuppetDB versions drop: this is being done for code simplification and because we cannot reasonably claim that we support versions older than 5.2 if we don't have automatic acceptance tests that verify that the app works with them and we (or at least I) don't have the capacity to manually test old versions.

Finally, this is a nice opportunity to upgrade Flask and other core dependencies. We will have a few pre-releases of this major app version so hopefully we will root out the bugs created during this upgrade in these beta tests.

### CODE REVIEW

**Comment by gdubicki** [comment]
We should probably combine it with #664...

File: README.md
Comment (kenyon) [inline README.md:29]:
```suggestion
* PuppetDB v. 5.2-7.10 (will most probably work with newer, but this has not been tested yet)
```

puppetboard works for me with puppetdb 7.10.

**Comment by gdubicki** [comment]
Hm, not sure why the tests are passing for me locally but not in GHA... Do I have to mock the response from the PuppetDB version endpoint for all the tests? 🤔

**Comment by gdubicki** [comment]
Please let me know what you think about this, @kenyon @bastelfreak @sebastianrakel @mterzo. Is the scope ok? Or should we add some other backward-incompatible change while we are doing major uptick? (Ping more people if needed, of course.)

File: puppetboard/utils.py
Comment (ekohl) [inline puppetboard/utils.py:46]: Nit: This is calling `parse(version)` quite a few times. Perhaps it should be stored in a variable?

Comment (gdubicki) [reply puppetboard/utils.py]: Thanks @ekohl 🙏, corrected. 😊

**Comment by gdubicki** [comment]
Please check out a pre-release [v4.0.0b1](https://github.com/voxpupuli/puppetboard/releases/tag/v4.0.0b1).

(Note: because of a PyPI issue I had to install it from the URL https://files.pythonhosted.org/packages/09/9d/5e59c13b3c9bb4ab250440b7fb1fd7385ece9213946983de41ac2e186ff0/puppetboard-4.0.0b1-py2.py3-none-any.whl )

**Comment by gdubicki** [comment]
Ping also @smortex and @buzzdeee as the maintainers of Puppetboard packages.

**Review by bastelfreak — APPROVED** [review]

**Comment by gdubicki** [comment]
I see no objections and I have an approval (thanks @bastelfreak!), so I am moving forward with this.

**Pull request merged**
